portal/models.py

```diff
@@ -410,6 +410,11 @@ def can_propose_project(
 
         return True, None
 
+    def get_expected_meetings(self, semester=Semester):
+        # Determine what kinds of meetings this student is expected to attend
+        meeting_types = [Meeting.LARGE_GROUP, Meeting.SMALL_GROUP, Meeting.WORKSHOP]
+        return Meeting.objects.filter(type__in=meeting_types, semester=semester)
```

Comment:
will this eventually vary based on student role?

Reply:
exactly!